flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -318,3 +319,82 @@ def move_dataset(dataset):
 
         waiting_for_shutdown.addCallback(move_dataset)
         return waiting_for_shutdown
+
+    @require_cluster(1)
+    def test_unregistered_volume(self, cluster):
+        """
+        If there is already a backend volume for a dataset when it is created,
+        that volume is used for that dataset.
+        """
+        api = get_backend_api(cluster.cluster_uuid)
+
+        # Create a volume for a dataset
+        dataset_id = uuid4()
+        volume = api.create_volume(dataset_id, size=get_default_volume_size())
+
+        # Then create the coresponding dataset.
+        wait_for_dataset = create_dataset(self, cluster, dataset_id=dataset_id)
+
+        def check_volumes(dataset):
+            [new_volume] = api.list_volumes()
```

Comment:
:eyes: the mutation makes it tricky, but putting the list in the assertEqual would
make the error messages much better if this expectation fails. You would
see what any excess volume was, instead of getting a "too many values to unpack"
error.